pyomo/contrib/piecewise/tests/test_triangulations.py

```diff
@@ -0,0 +1,222 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+
+import itertools
+from unittest import skipUnless
+import pyomo.common.unittest as unittest
+from pyomo.contrib.piecewise.triangulations import (
+    get_unordered_j1_triangulation,
+    get_ordered_j1_triangulation,
+    get_Gn_hamiltonian,
+    get_grid_hamiltonian,
+)
+from pyomo.common.dependencies import numpy as np, numpy_available
+from math import factorial
+import itertools
+
+
+class TestTriangulations(unittest.TestCase):
+
+    # check basic functionality for the unordered j1 triangulation.
+    @unittest.skipUnless(numpy_available, "numpy is not available")
+    def test_J1_small(self):
+        points = [
+            [0.5, 0.5],  # 0
+            [0.5, 1.5],  # 1
+            [0.5, 2.5],  # 2
+            [1.5, 0.5],  # 3
+            [1.5, 1.5],  # 4
+            [1.5, 2.5],  # 5
+            [2.5, 0.5],  # 6
+            [2.5, 1.5],  # 7
+            [2.5, 2.5],  # 8
+        ]
+        triangulation = get_unordered_j1_triangulation(points, 2)
+        self.assertTrue(
+            np.array_equal(
+                triangulation.simplices,
+                np.array(
+                    [
+                        [0, 1, 4],
+                        [1, 2, 4],
+                        [4, 6, 7],
+                        [4, 7, 8],
+                        [0, 3, 4],
+                        [2, 4, 5],
+                        [3, 4, 6],
+                        [4, 5, 8],
+                    ]
+                ),
+            )
+        )
+
+    def check_J1_ordered(self, points, num_points, dim):
+        ordered_triangulation = get_ordered_j1_triangulation(points, dim).simplices
+        self.assertEqual(
+            len(ordered_triangulation), factorial(dim) * (num_points - 1) ** dim
+        )
+        for idx, first_simplex in enumerate(ordered_triangulation):
+            if idx != len(ordered_triangulation) - 1:
+                second_simplex = ordered_triangulation[idx + 1]
+                # test property (2) which also guarantees property (1)
```

Comment:
These are properties (1) and (2) from Vielma et al. 2010?

Reply:
Yes